algorithms/unit_tests/TestRandom.hpp

```diff
@@ -424,6 +432,12 @@ struct test_random_scalar {
       }
 #endif
 
+#if defined(KOKKOS_BHALF_T_IS_FLOAT) && !KOKKOS_BHALF_T_IS_FLOAT
+      if (std::is_same<Scalar, Kokkos::Experimental::bhalf_t>::value) {
+        variance_factor = 15.01;
```

Comment:
Where does this number come from?

Reply:
Based on the test failure output with a smaller variance factor, this is approximately the smallest variance factor I found that allows the tests to pass.